genesis/experiment/member/member.go

```diff
@@ -17,24 +17,85 @@
 package member
 
 import (
+	"github.com/insolar/insolar/core"
+	"github.com/insolar/insolar/cryptohelpers/ecdsa"
+	"github.com/insolar/insolar/genesis/experiment/member/signer"
+	"github.com/insolar/insolar/genesis/proxy/member"
+	"github.com/insolar/insolar/genesis/proxy/rootdomain"
+	"github.com/insolar/insolar/genesis/proxy/wallet"
 	"github.com/insolar/insolar/logicrunner/goplugin/foundation"
 )
 
 type Member struct {
 	foundation.BaseContract
 	Name      string
-	PublicKey []byte
+	PublicKey string
 }
 
 func (m *Member) GetName() string {
 	return m.Name
 }
-func (m *Member) GetPublicKey() []byte {
+func (m *Member) GetPublicKey() string {
 	return m.PublicKey
 }
 
-func New(name string) *Member {
+func New(name string, key string) *Member {
 	return &Member{
-		Name: name,
+		Name:      name,
+		PublicKey: key,
 	}
 }
+
+func (m *Member) AuthorizedCall(ref string, method string, params []interface{}, seed []byte, sign []byte) ([]interface{}, *foundation.Error) {
+	serialized, err := signer.Serialize(ref, method, params, seed)
+	if err != nil {
+		return nil, &foundation.Error{S: err.Error()}
+	}
+	verified, err := ecdsa.Verify(serialized, sign, m.PublicKey)
+	if err != nil {
+		return nil, &foundation.Error{S: err.Error()}
+	}
+	if !verified {
+		return nil, &foundation.Error{S: "Incorrect signature"}
+	}
+
+	switch method {
+	case "CreateMember":
+		domain := rootdomain.GetObject(core.NewRefFromBase58(ref))
+		name, ok := params[0].(string)
+		if !ok {
+			return nil, &foundation.Error{S: "First parameter must be string"}
+		}
+		key, ok := params[1].(string)
+		if !ok {
+			return nil, &foundation.Error{S: "Second parameter must be string"}
+		}
+		return []interface{}{domain.CreateMember(name, key)}, nil
+	case "GetName":
+		membr := member.GetObject(core.NewRefFromBase58(ref))
+		return []interface{}{membr.GetName()}, nil
+	case "GetPublicKey":
+		membr := member.GetObject(core.NewRefFromBase58(ref))
+		return []interface{}{membr.GetPublicKey()}, nil
+	case "GetBalance":
+		wallet := wallet.GetImplementationFrom(core.NewRefFromBase58(ref))
+		return []interface{}{wallet.GetTotalBalance()}, nil
+	case "SendMoney":
+		wallet := wallet.GetImplementationFrom(core.NewRefFromBase58(ref))
+		amount, ok := params[0].(uint64)
+		if !ok {
+			return nil, &foundation.Error{S: "First parameter must be uint"}
+		}
+		to, ok := params[1].(string)
+		if !ok {
+			return nil, &foundation.Error{S: "Second parameter must be string"}
+		}
+		v := core.NewRefFromBase58(to)
+		wallet.Transfer(uint(amount), &v)
+		return nil, nil
+	case "DumpAllUsers":
+		domain := rootdomain.GetObject(core.NewRefFromBase58(ref))
+		return []interface{}{domain.DumpAllUsers()}, nil
+	}
```

Comment:
We also have `"DumpUserInfo"` method

Reply:
I removed "DumpAllUsers" also, because these methods should be modified for work via AuthorizedCall